systemtest/benchtest/eventhandler/handler.go

```diff
@@ -128,7 +130,7 @@ func New(p string, t *Transport, storage fs.FS, warmup uint) (*Handler, error) {
 
 // SendBatches sends the loaded trace data to the configured transport. Returns
 // the total number of documents sent and any transport errors.
-func (h *Handler) SendBatches(ctx context.Context) (uint, error) {
+func (h *Handler) SendBatches(ctx context.Context, l *rate.Limiter) (uint, error) {
```

Comment:
Do you think it would be reasonable to keep this signature as is, and instead inject `*rate.Limiter` when the `Handler` is constructed in `event handler.New`? It seems unlikely that we'll use different rate limiters on different calls to `SendBatches`.

Reply:
`WarmupServer` is also defined on the event handler receiver and currently I don't limit warmup events by using a Inf limiter. Having the limiter in the handler but overriding the limiter in warmup server might lead to confusion so I kept limiter as an argument.

If we limit warmup events also then I think having Limiter as part of the handler struct would make more sense but I am not sure if we should limit warmup events. WDYT?

Reply:
I see, I am more inclined to have a consistent limit both in warm-up and benchmark execution rather than warm up the server as fast as we can and then rate-limit on the benchmarks.

I can't think of a use-case where we'd want to warm up the APM Server by sending events events at max throughput and then wanting to benchmark a sustained throughput.

Also because of the way the flags are configured, I would expect the apmbench to limit both warm up and each of the benchmark cases.

Reply:
> I can't think of a use-case where we'd want to warm up the APM Server by sending events events at max throughput and then wanting to benchmark a sustained throughput.

Hmm, how about the case where we want to benchmark a low throughput. Having a lower rate limit for warmup events might take too much time since warmup events have a high count to be useful.

Reply:
Looking at the code again, it seems that warmup initialises a separate event handler so promoting rate.Limiter to the handler struct sounds good, I will do the changes. WDYT about above point? Should we have a separate limiter for warmup events to make warmup quickly?

Reply:
sounds good, I was thinking if it would make sense to introduce a flag, but I'd rather not unless it becomes a problem. It should be a straightforward enough change to do if we need to, and we can already control the number of warmup events. Does that sound reasonable to you?

Reply:
> I was thinking if it would make sense to introduce a flag, but I'd rather not unless it becomes a problem.

I was thinking along the same lines, agreed on introducing it if required. For now I will make the warmup events use the  limiter with same configuration.

Reply:
Updated the code with the required changes